src/async_fuse/memfs/inode.rs

```diff
@@ -0,0 +1,116 @@
+use std::sync::Arc;
+use clippy_utilities::OverflowArithmetic;
+use tokio::sync::Mutex;
+use crate::async_fuse::fuse::protocol::INum;
+use crate::async_fuse::memfs::{dist, S3MetaData};
+use crate::async_fuse::memfs::dist::etcd;
+use crate::async_fuse::memfs::s3_wrapper::S3BackEnd;
+use crate::common::etcd_delegate::EtcdDelegate;
+
+#[derive(Debug)]
+pub(crate) struct InodeState {
+    range_begin_end:Mutex<(INum,INum)>,
+    recycle_unused:crossbeam_queue::SegQueue<INum>,
+}
+
+impl InodeState {
+    pub fn new()->Self{
+        Self{
+            range_begin_end:Mutex::new((0,0)),
+            recycle_unused: Default::default(),
+        }
+    }
+}
+
+impl InodeState {
+    /// just get a unique inum
+    async fn alloc_inum(&self, etcd_client:&Arc<EtcdDelegate>) -> INum {
+        const INODE_RANGE:u64=10000;
+        if let Some(inum)= self.recycle_unused.pop(){
+            return inum;
+        }
+        let mut locked=self.range_begin_end.lock().await;
+        if locked.0==locked.1{
+            // need update
+            let ret=etcd::fetch_add_inode_next_range(
+                // here appeared an await, so we need to use tokio::sync::Mutex
+                Arc::clone(etcd_client),INODE_RANGE).await.unwrap_or_else(|e|{
+                panic!("failed to fetch add inode next range, error is {:?}", e)
+            });
+            // begin
+            locked.0=ret;
+            // end
+            locked.1=ret.overflow_add(INODE_RANGE-1);
+            ret
+        }else {
+            let ret=locked.0;
+            locked.0=locked.0.overflow_add(1);
+            ret
+        }
+    }
+
```

Comment:
please fmt the code, removing the unnessary blank lines.

Reply:
Ok.